src/ker_horner_allw_loop_constexpr.h

```diff
@@ -1,234 +1,235 @@
-// Code generated by gen_all_horner_C_code.m in finufft/devel
-// Authors: Alex Barnett & Ludvig af Klinteberg.
-// (C) The Simons Foundation, Inc.
+// Header of static arrays of monomial coeffs of spreading kernel function in each
+// fine-grid interval. Generated by gen_all_horner_cpp_header.m in finufft/devel
+// Authors: Alex Barnett, Ludvig af Klinteberg, Marco Barbone & Libin Lu.
+// (C) 2018--2024 The Simons Foundation, Inc.
 #include <array>
-template<uint8_t w> constexpr auto nc200() noexcept { return w + 2 + (w <= 8); }
+
+template<uint8_t w> constexpr auto nc200() noexcept {
+  constexpr uint8_t ncs[] = {4, 6, 7, 8, 9, 10, 10, 11, 12, 13, 14, 15, 16, 17, 18, };
+  return ncs[w-2];
+}
 
 template<class T, uint8_t w>
 constexpr std::array<std::array<T, w>, nc200<w>()> get_horner_coeffs_200() noexcept {
```

Comment:
to remove nc200 it would be possible by doing:
`constexpr auto get_horner_coeffs_200() noexcept {`
so the return type is auto and then: 
``` cpp
     return std::array<std::array<T, w>, 4> {{ // hardcoded nc value
          {-2.5024197515954172E-01, 2.5024197515954183E-01},
          {-2.2584311526143722E-02, -2.2584311526143454E-02},
          {7.0481840008800800E-01, -7.0481840008800789E-01},
          {5.5428559551548429E-01, 5.5428559551548384E-01},
      }}
```
for each block.

`spreadinterp.cpp` should then be changes so that if gets the number of nc using `horner_coeffs.size()` which should be constexpr and potentially it should work with no issues

Reply:
Also,

the invalid case (after w==16) needs to be changed from:
```
...
  } else {
    static_assert(w >= 2, "w must be >= 2");
    static_assert(w <= 16, "w must be <= 16");
    return {};
  }
};
```
to
```cpp
...
 }
  static_assert(w >= 2, "w must be >= 2");
  static_assert(w <= 16, "w must be <= 16");
};
```
the static assert are not in any branch (executed all the time at compile time) and there is not empty return type as returning {} on auto is supported by gcc but not clang/msvc.